counsel.el

```diff
@@ -5222,37 +5222,37 @@ subdirectories that builds may be invoked in."
            counsel-compile-build-directories))
 
 (defun counsel--get-build-subdirs (blddir)
-  "Return all subdirs of BLDDIR sorted by access time."
+  "Return all subdirs of BLDDIR sorted by modification time."
   (mapcar #'car (sort (directory-files-and-attributes
                        blddir t directory-files-no-dot-files-regexp t)
                       (lambda (x y)
-                        ;; FIXME: Access time is NOT the 7th file attribute.
                         (time-less-p (nth 6 y) (nth 6 x))))))
 
 (defun counsel-compile-get-build-directories (&optional dir)
   "Return a list of potential build directories."
   (let* ((srcdir (or dir (counsel--compile-root)))
-         (blddir (counsel--find-build-subdir srcdir))
-         (props `(srcdir ,srcdir blddir ,blddir recursive t)))
+         (blddir (counsel--find-build-subdir srcdir)))
     (mapcar (lambda (s)
-              (setq s (concat (propertize "Select build in "
-                                          'face 'font-lock-warning-face)
-                              (propertize s 'face 'dired-directory)))
-              (add-text-properties 0 (length s) props s)
-              s)
+              (propertize
+               (concat (propertize "Select build in "
+                                   'face 'font-lock-warning-face)
+                       (propertize s 'face 'dired-directory))
+               'blddir s 'srcdir srcdir 'recursive t))
             (and blddir (counsel--get-build-subdirs blddir)))))
 
-;; No easy way to make directory local, would buffer local make more sense?
+;; This is a workaround for the fact there is no concept of "project"
+;; local variables (as opposed to for example buffer-local). So we
```

Comment:
Nitpick: Please honour the `sentence-end-double-space` setting in `dir-locals-file` and end sentences with two spaces.